ruma-common/src/push.rs

```diff
@@ -925,4 +1009,229 @@ mod tests {
+                }
+            }"#,
+        )
+        .unwrap();
+        let mut actions = set.get_actions(&notice, context_one_to_one);
+        assert_matches!(actions.next().unwrap(), Action::DontNotify);
+
+        let at_room = serde_json::from_str::<Raw<JsonValue>>(
+            r#"{
+                "type": "m.room.message",
+                "sender": "@rantanplan:server.name",
+                "content": {
+                    "body": "@room Attention please!",
+                    "msgtype": "m.text"
+                }
+            }"#,
+        )
+        .unwrap();
+        let mut actions = set.get_actions(&at_room, context_public_room);
+        assert_matches!(actions.next().unwrap(), Action::Notify);
+        assert_matches!(actions.next().unwrap(), Action::SetTweak(Tweak::Highlight(true)));
+
+        let empty = serde_json::from_str::<Raw<JsonValue>>(r#"{}"#).unwrap();
+        let mut actions = set.get_actions(&empty, context_one_to_one);
+        assert!(actions.next().is_none());
+    }
+
+    #[test]
+    fn custom_ruleset_applies() {
+        let context_one_to_one = &PushConditionRoomCtx {
+            room_id: "!dm:server.name".into(),
+            member_count: 2u32.into(),
+            user_display_name: "Jolly Jumper".into(),
+            users_power_levels: BTreeMap::new(),
+            default_power_level: 50.into(),
+            notification_power_levels: NotificationPowerLevels { room: 50.into() },
+        };
+
+        let message = serde_json::from_str::<Raw<JsonValue>>(
+            r#"{
+                "sender": "@rantanplan:server.name",
+                "type": "m.room.message",
+                "content": {
+                    "msgtype": "m.text",
+                    "body": "Great joke!"
+                }
+            }"#,
+        )
+        .unwrap();
+
+        let mut set = Ruleset::new();
+        let disabled = AnyPushRule::Underride(ConditionalPushRule {
+            actions: vec![Action::Notify],
+            default: false,
+            enabled: false,
+            rule_id: "disabled".into(),
+            conditions: vec![PushCondition::RoomMemberCount {
+                is: RoomMemberCountIs::from(uint!(2)),
+            }],
+        });
+        set.add(disabled);
+
+        let test_set = set.clone();
+        let mut actions = test_set.get_actions(&message, context_one_to_one);
+        assert!(actions.next().is_none());
+
+        let no_conditions = AnyPushRule::Underride(ConditionalPushRule {
+            actions: vec![Action::SetTweak(Tweak::Highlight(true))],
+            default: false,
+            enabled: true,
+            rule_id: "no.conditions".into(),
+            conditions: vec![],
+        });
+        set.add(no_conditions);
+
+        let test_set = set.clone();
+        let mut actions = test_set.get_actions(&message, context_one_to_one);
+        assert_matches!(actions.next().unwrap(), Action::SetTweak(Tweak::Highlight(true)));
```

Comment:
Does this work or is there a missing `Debug` impl for one of the types involved?

```suggestion
        assert_matches!(actions.next(), Some(Action::SetTweak(Tweak::Highlight(true))));
```

Reply:
It does work. I'm gonna change this.